mypy/checker.py

```diff
@@ -2693,12 +2723,31 @@ def flatten(t: Expression) -> List[Expression]:
 
 
 def get_isinstance_type(expr: Expression, type_map: Dict[Expression, Type]) -> Type:
-    type = type_map[expr]
+    all_types = [type_map[e] for e in flatten(expr)]
+
+    types = []  # type: List[Type]
+
+    for type in all_types:
+        if isinstance(type, FunctionLike):
+            if type.is_type_obj():
+                # Type variables may be present -- erase them, which is the best
+                # we can do (outside disallowing them here).
+                type = erase_typevars(type.items()[0].ret_type)
```

Comment:
I should probably do it in a separate PR though? (Since I deleted my function, your comment now applies only to `get_isinstance_type`, which I'm not modifying.) Or is it ok to incorporate it here as well?